# Week 1: Introduction to Data

In 2011, researchers at the Texas Transportation Institute and the Center for Transportation Research at UT Austin ran an advertising campaign aimed at recruiting Austin cyclists to join the South Congress Bike Mapping Project.  As members of the project, cyclists downloaded and used Cycle Tracks, a smartphone app developed by the San Francisco County Transportation Authority to track where people are riding their bikes based on their GPS points.  The goal was to gain new information about bike commuting patterns and this data set is based on the results of the study: 3600 trips tracked from 315 users over a 6 month period.   Data includes distances traveled, speed of travel, and reasons for travel among other variables.

In [1]:
import pandas as pd
import numpy as np

bike = pd.read_csv("BikeData.csv")

In [2]:
bike

Unnamed: 0  user_id  age gender  student  employed  \
0             1        1   28      M        1         1   
1             2        2   35      M        0         1   
2             3        3   28      M        0         1   
3             4        4   44      F        0         1   
4             5        5   42      M        0         1   
5             6        6   36      M        0         1   
6             7        7   45      M        0         1   
7             8        8   54      M        0         1   
8             9        9   39      M        0         1   
9            10       10   44      M        0         0   
10           11       11   29      M        0         1   
11           12       12   32      M        0         1   
12           13       13   40      M        0         1   
13           14       14   29      M        0         1   
14           15       15   26      M        1         1   
15           16       16   39      M        0         1   
16           17       17   33      M        0         1   
17           18       18   32      F        1         1   
18           19       19   43      M        0         1   
19           20       20   29      M        0         0   
20           21       21   51      F        0         1   
21           22       22   36      M        0         1   
22           23       23   29      M        0         1   
23           24       24   30      F        0         1   
24           25       25   30      M        0         1   
25           26       26   35      M        0         1   
26           27       27   27      F        0         1   
27           28       28   37      M        0         1   
28           29       29   27      F        0         1   
29           30       30   25      M        0         1   
..          ...      ...  ...    ...      ...       ...   
91           92       92   38      M        0         1   
92           93       93   29      M        0         1   
93           94       94   36      F        0         1   
94           95       95   29      M        0         1   
95           96       96   33      M        0         1   
96           97       97   29      M        1         0   
97           98       98   37      M        0         1   
98           99       99   43      M        0         1   
99          100      100   30      M        0         1   
100         101      101   40      M        0         1   
101         102      102   28      M        0         0   
102         103      103   25      F        1         0   
103         104      104   44      M        0         1   
104         105      105   42      F        0         1   
105         106      106   33      M        0         1   
106         107      107   37      M        0         1   
107         108      108   32      M        0         1   
108         109      109   51      F        0         1   
109         110      110   54      M        0         1   
110         111      111   30      M        1         0   
111         112      112   34      M        0         1   
112         113      113   34      M        0         1   
113         114      114   43      M        0         1   
114         115      115   33      F        1         1   
115         116      116   29      F        0         0   
116         117      117   46      M        0         1   
117         118      118   41      F        0         1   
118         119      119   44      F        0         1   
119         120      120   46      F        0         1   
120         121      121   38      M        0         1   

                    cyc_freq  distance  time  speed  
0                      Daily      3.25    15  13.00  
1                      Daily      1.11     5  13.32  
2                      Daily      5.59    23  14.58  
3     Less than once a month      3.24    24   8.10  
4     Several times per week      7.81    26  18.02  
5     Several times pe

### Pre Lab
In this Pre Lab, we will ask a question about the bike dataset:  

How many of the cyclists were students, how often did they ride, and what was the average distance they rode?

In [3]:
# What is the age of the 7th rider in the dataset?
bike["age"][bike["user_id"] == 7]

6    45
Name: age, dtype: int64

In [4]:
# How many of the first 10 riders in the dataset ride daily?
bike_10 = bike[0:10]
bike_10_valuecounts = bike_10["cyc_freq"].value_counts()
bike_10_valuecounts["Daily"]

3

In [5]:
# What is the speed of the first female who cycles less than one time per 
# month (in miles/hour)?
bike_female = bike[bike["gender"] == "F"]
bike_female_speed = bike_female["speed"][bike["cyc_freq"] == "Less than once a month"]
print bike_female_speed.keys()
print bike_female_speed[3]

Int64Index([3, 56], dtype='int64')
8.1


1. What type of variable is student?
Categorical
2. What type of variable is cyc_freq?
Categorical
3. What type of variable is distance?
Numerical

In [6]:
# 1. Find the number of students in the dataset.
print bike["student"].sum()

14


In [7]:
# 2. Pull out the student data into a separate dataframe for analysis.
bike_student = bike[bike["student"] == 1]

In [8]:
# 3. Make a table to find how often the students ride.
bike_student_value_counts = bike_student["cyc_freq"].value_counts()
bike_student_value_counts

Daily                     8
Several times per week    6
Name: cyc_freq, dtype: int64

In [9]:
# 4. Find the average distance ridden.
print bike_student["distance"].mean()

6.25785714286


In this Pre Lab we examined data on 14 student riders.  Most of the students riders (a total of 8 out of 14) rode their bikes daily.  On average, the students rode about 6.26 miles on each trip.

### Lab

In [10]:
# How many daily riders are in the dataset?
bike_cyc_freq_value_counts = bike["cyc_freq"].value_counts()
bike_cyc_freq_value_counts["Daily"]

47

In [11]:
# How many of the daily riders are male?
bike_daily = bike[bike["cyc_freq"] == "Daily"]
bike_daily["gender"].value_counts()

M    38
F     9
Name: gender, dtype: int64

In [12]:
# What is the average age of daily riders?
bike_daily["age"].mean()

33.659574468085104

In [13]:
# What is the average age of the female daily riders
bike_daily_female = bike_daily[bike_daily["gender"] == "F"]
bike_daily_female["age"].mean()

32.55555555555556

In [14]:
# What is the average age of the male daily riders?
bike_daily_male = bike_daily[bike_daily["gender"] == "M"]
bike_daily_male["age"].mean()

33.921052631578945

In [15]:
# How would you create a datafile that only includes male daily riders that 
# are age 30 or older? 
bike_daily_above_30 = bike_daily[bike_daily["age"] >= 30]

In [16]:
# How many daily male riders are age 30 or older?
bike_daily_above_30["gender"].value_counts()

M    25
F     5
Name: gender, dtype: int64

In this lab, we examined data on 47 daily riders.  Most of the daily riders (a total of 38) were maile.  The average age of all daily rides was 33.7.  The average age of female daily riders was 32.6, while the average age of male daily riders was 33.7.